File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.java
class Solution {

   boolean isFind(int[] a, int target) {
        int left = 0;
        int right = a.length - 1;

        while (left <= right) {
            int mid = left + (right - left) / 2;

            if (a[mid] == target) {
                return true;  
            } else if (a[mid] < target) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }
        return false; 
    }

    public boolean searchMatrix(int[][] matrix, int target) {
        for(int a[]:matrix){
            if(isFind(a,target)){
                return true;
            }
        }
        return false;
    }
}

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        for a in matrix:
            if self.Find(a,target):
                return True
        return False
    def Find(self,a:List[int],t:int)->bool:
        l=0
        r=len(a)-1
        while(l<=r):
            m=l+(r-l)//2
            if(a[m]==t):
                return True
            if(a[m]<t):
                l=m+1
            else:
                r=m-1
        return False
        